File: Client/src/app/features/address/address.model.ts
export class Address {
    public streetName: string;
    public number: number;
    public neighborhood: string;
    public city: string;
    public state: string;
    public country: string;

    constructor (entity: any) {
        this.streetName = entity.streetName;
        this.number = entity.number;
        this.neighborhood = entity.neighborhood ;
        this.city = entity.city;
        this.state = entity.state;
        this.country = entity.country;
    }
}

File: Server/Projeto_NFe.Infra.XML/Features/Addresses/AddressModel.cs
using System.Diagnostics.CodeAnalysis;
using System.Xml.Serialization;

namespace Projeto_NFe.Infra.XML.Features.Addresses
{
    [ExcludeFromCodeCoverage]
    public class AddressModel
    {
        [XmlElement(ElementName = "xLgr")]
        public string StreetName { get; set; }

        [XmlElement(ElementName = "nro")]
        public int Number { get; set; }

        [XmlElement(ElementName = "xNeighborhood")]
        public string Neighborhood { get; set; }

        [XmlElement(ElementName = "xMun")]
        public string City { get; set; }

        [XmlElement(ElementName = "UF")]
        public string State { get; set; }

        [XmlElement(ElementName = "xCountry")]
        public string Country { get; set; }
    }
}
